pkg/webhook/resourcedistribution/validating/resourcedistribution_create_update_handler.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2021 The Kruise Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validating
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/klog"
+	coreval "k8s.io/kubernetes/pkg/apis/core/validation"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// ResourceDistributionCreateUpdateHandler handles ResourceDistribution
+type ResourceDistributionCreateUpdateHandler struct {
+	Client client.Client
+
+	// Decoder decodes objects
+	Decoder *admission.Decoder
+}
+
+var _ admission.Handler = &ResourceDistributionCreateUpdateHandler{}
+
+// validateResourceDistributionSpec validate Spec when creating and updating
+// (1). validate resource itself
+// (2). validate targets
+func (h *ResourceDistributionCreateUpdateHandler) validateResourceDistributionSpec(obj, oldObj *appsv1alpha1.ResourceDistribution, fldPath *field.Path) (allErrs field.ErrorList) {
```

Comment:
Is it necessary to determine the conflict between multiple ResourceDistribution, e.g. resources of the same kind, name can only belong to one ResourceDistribution?

Reply:
we will check this conflict in the controller.
actually, we do that in the last version, but we remove this conflict detection because parsing target namespaces and detecting this conflict are too costly in webhooks.

Reply:
After many considerations, I removed this conflict detection logic because it will restrict use in many scenarios. For example, users want to configure configMaps with the same name but different data for different environments, or in multi-user situations, the behavior of one user will affect the use of other users.